include/bm/storage/bm_storage_backend.h

```diff
@@ -24,57 +24,57 @@ extern "C" {
  *
  * @note This function must be defined by the backend.
  *
- * @retval NRF_SUCCESS on success.
- * @retval NRF_ERROR_BUSY If the implementation-specific resource is busy.
- * @retval NRF_ERROR_INTERNAL If an implementation-specific internal error occurred.
+ * @retval 0 on success.
+ * @retval -EBUSY If the implementation-specific resource is busy.
+ * @retval -EIO If an implementation-specific internal error occurred.
  */
-uint32_t bm_storage_backend_init(struct bm_storage *storage);
+int bm_storage_backend_init(struct bm_storage *storage);
 /**
  * @brief Uninitialize the storage peripheral.
  *
  * @note This function is optional. If not defined in the backend, a weak implementation will return
- * NRF_ERROR_NOT_SUPPORTED.
+ * -ENOTSUP.
  *
- * @retval NRF_SUCCESS on success.
- * @retval NRF_ERROR_BUSY If the implementation-specific backend is busy with an ongoing operation.
- * @retval NRF_ERROR_NOT_SUPPORTED If the backend does not support uninitialization.
+ * @retval 0 on success.
+ * @retval -EBUSY If the implementation-specific backend is busy with an ongoing operation.
+ * @retval -ENOTSUP If the backend does not support uninitialization.
  */
-uint32_t bm_storage_backend_uninit(struct bm_storage *storage);
+int bm_storage_backend_uninit(struct bm_storage *storage);
 /**
  * @brief Read data from non-volatile memory.
  *
  * @note This function must be defined by the backend.
  *
- * @retval NRF_SUCCESS on success.
- * @retval NRF_ERROR_FORBIDDEN If the implementation-specific backend has not been initialized.
+ * @retval 0 on success.
+ * @retval -EPERM If the implementation-specific backend has not been initialized.
  */
-uint32_t bm_storage_backend_read(const struct bm_storage *storage, uint32_t src, void *dest,
-				 uint32_t len);
+int bm_storage_backend_read(const struct bm_storage *storage, uint32_t src, void *dest,
+			    uint32_t len);
 /**
  * @brief Write bytes to non-volatile memory.
  *
  * @note This function must be defined by the backend.
  *
- * @retval NRF_SUCCESS on success.
- * @retval NRF_ERROR_FORBIDDEN If the implementation-specific backend has not been initialized.
- * @retval NRF_ERROR_BUSY If the implementation-specific backend is busy with an ongoing operation.
- * @retval NRF_ERROR_INTERNAL If an implementation-specific internal error occurred.
+ * @retval 0 on success.
+ * @retval -EPERM If the implementation-specific backend has not been initialized.
+ * @retval -EBUSY If the implementation-specific backend is busy with an ongoing operation.
+ * @retval -EIO If an implementation-specific internal error occurred.
  */
-uint32_t bm_storage_backend_write(const struct bm_storage *storage, uint32_t dest, const void *src,
-				  uint32_t len, void *ctx);
+int bm_storage_backend_write(const struct bm_storage *storage, uint32_t dest, const void *src,
```

Comment:
This can return `-EFAULT` from the SoftDevice backend if `src` is not 32-bit aligned. Not documented here.